From 3fc0a9ce3dc97c879222be4ee92561757e7e9379 Mon Sep 17 00:00:00 2001
From: Guillaume Gardet <guillaume.gardet@arm.com>
Date: Wed, 29 Apr 2020 12:03:44 +0200
Subject: [PATCH] Fix aarch64/arm detection

Upstream-Status: Backport [v2.35.0 https://github.com/IntelRealSense/librealsense/commit/6a61d5a946c84d37266a08bbfd90e34849e13517]

* with OE builds there is e.g.:
  librealsense2/2.34.0-1-r0/git$ $CC -dumpmachine OUTPUT_VARIABLE MACHINE
  aarch64-oe-linux
  which causes that neither option does apply and it adds the default
  -mssse3 from the "else" branch causing:

Determining if the include file pthread.h exists failed with the following output:
Change Dir: /jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/build/CMakeFiles/CMakeTmp

Run Build Command(s):ninja cmTC_74201 && [1/2] Building C object CMakeFiles/cmTC_74201.dir/CheckIncludeFile.c.o
FAILED: CMakeFiles/cmTC_74201.dir/CheckIncludeFile.c.o
/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot-native/usr/bin/aarch64-oe-linux/aarch64-oe-linux-gcc   -mcpu=cortex-a72+crc+crypto  --sysroot=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0=/usr/src/debug/librealsense2/2.34.0-1-r0                      -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0=/usr/src/debug/librealsense2/2.34.0-1-r0                      -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot=                      -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot-native=   -mcpu=cortex-a72+crc+crypto  --sysroot=/jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot   -pedantic -g -D_DEFAULT_SOURCE   -mssse3 -pthread  -fPIE -o CMakeFiles/cmTC_74201.dir/CheckIncludeFile.c.o   -c CheckIncludeFile.c
aarch64-oe-linux-gcc: error: unrecognized command line option '-mssse3'
ninja: build stopped: subcommand failed.

* it also breaks glfw detection and then it fails with missing Threads
  -- GLFW 3.3 not found; using internal version
  CMake Error at /jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
    Could NOT find Threads (missing: Threads_FOUND)
  Call Stack (most recent call first):
    /jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
    /jenkins/mjansa/build/ros/ros2-foxy-dunfell/tmp-glibc/work/aarch64-oe-linux/librealsense2/2.34.0-1-r0/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindThreads.cmake:220 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    third-party/glfw/CMakeLists.txt:63 (find_package)

---
 CMake/unix_config.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMake/unix_config.cmake b/CMake/unix_config.cmake
index 0de80e3..7e794cc 100644
--- a/CMake/unix_config.cmake
+++ b/CMake/unix_config.cmake
@@ -7,11 +7,11 @@ macro(os_set_flags)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-switch -Wno-multichar -Wsequence-point -Wformat -Wformat-security")
 
     execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpmachine OUTPUT_VARIABLE MACHINE)
-    if(${MACHINE} MATCHES "arm-linux-gnueabihf")
+    if(${MACHINE} MATCHES "arm-*")
         set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mfpu=neon -mfloat-abi=hard -ftree-vectorize -latomic")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=neon -mfloat-abi=hard -ftree-vectorize -latomic")
         add_definitions(-DRASPBERRY_PI)
-    elseif(${MACHINE} MATCHES "aarch64-linux-gnu")
+    elseif(${MACHINE} MATCHES "aarch64-*")
         set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mstrict-align -ftree-vectorize")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mstrict-align -ftree-vectorize")
     elseif(${MACHINE} MATCHES "powerpc64(le)?-linux-gnu")
@@ -21,7 +21,7 @@ macro(os_set_flags)
         set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mssse3")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mssse3")
         set(LRS_TRY_USE_AVX true)
-    endif(${MACHINE} MATCHES "arm-linux-gnueabihf")
+    endif(${MACHINE} MATCHES "arm-*")
 
     if(BUILD_WITH_OPENMP)
         find_package(OpenMP REQUIRED)
